processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -57,12 +55,13 @@ public ExtractionFilter(
     if (allDimVals != null) {
```

Comment:
Let's adjust this if statement to build a new Indexed instead...  So, something like

```
if (allDimVals == null) {
  allDimVals = new Indexed<String>(){
    public int size() { return 1; }
    public String get(int index) { return null }
  };
} 

for (int i = 0; i < allDimVals.size(); ++i) {
  ...
}
```